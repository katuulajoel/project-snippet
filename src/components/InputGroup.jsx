import PropTypes from "prop-types";
import React from "react";

import Input from "./Input";
import { addPropsToChildren } from "./utils/children";
import { filterEventProps } from "./utils/events";
import { filterInputProps } from "./utils/forms";

const InputGroup = (props) => {
  const {
    className,
    prepend,
    isPrependText,
    type,
    size,
    placeholder,
    append,
    isAppendText,
    appendFunc,
    value,
  } = props;
  return (
    <div className={`input-group ${className || ""}`}>
      {prepend ? (
        <span className="input-group-prepend">
          {isPrependText ? (
            <span className="input-group-text">{prepend}</span>
          ) : (
            prepend
          )}
        </span>
      ) : null}
      <Input
        type={type}
        className="form-control"
        size={size}
        placeholder={placeholder}
        {...filterInputProps(props)}
        {...filterEventProps(props)}
      />
      {append && (
        <span className="input-group-prepend">
          {isAppendText ? (
            <span className="input-group-text">{append}</span>
          ) : appendFunc ? (
            value !== "" &&
            addPropsToChildren(append, {
              onClick: () => appendFunc(this),
            })
          ) : (
            append
          )}
        </span>
      )}
    </div>
  );
};

InputGroup.defaultProps = {
  isPrependText: true,
  isAppendText: true,
};

InputGroup.propTypes = {
  type: PropTypes.string,
  className: PropTypes.string,
  placeholder: PropTypes.string,
  prepend: PropTypes.object,
  append: PropTypes.object,
  isPrependText: PropTypes.bool,
  isAppendText: PropTypes.bool,
  appendFunc: PropTypes.func,
  size: PropTypes.string,
  onChange: PropTypes.func,
  value: PropTypes.string,
};

export default InputGroup;
