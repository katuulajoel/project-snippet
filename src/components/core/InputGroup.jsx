import PropTypes from "prop-types";
import React from "react";
import IconButton from "./IconButton";

import Input from "./Input";
import { addPropsToChildren } from "./utils/children";
import { filterEventProps } from "./utils/events";
import { filterInputProps } from "./utils/forms";

export default class InputGroup extends React.Component {
    static defaultProps = {
        isPrependText: true,
        isAppendText: true
    };

    static propTypes = {
        type: PropTypes.string,
        className: PropTypes.string,
        placeholder: PropTypes.string,
        prepend: PropTypes.object,
        append: PropTypes.object,
        isPrependText: PropTypes.bool,
        isAppendText: PropTypes.bool,
        appendFunc: PropTypes.func,
        size: PropTypes.string,
        onFocus: PropTypes.func,
        onChange: PropTypes.func,
        value: PropTypes.string
    };

    constructor(props) {
        super(props);
        this.state = { focus: false };
    }

    onChangeFocus(focus) {
        this.setState({ focus });
        if (this.props.onFocus) {
            this.props.onFocus();
        }
    }

    appendtest = context => {
        console.log(context);
        return (
            <IconButton
                onClick={() => {
                    console.log("you click the search box");
                }}
                name="x-circle"
            />
        );
    };

    render() {
        return (
            <div
                className={`input-group ${this.props.className || ""} ${
                    this.state.focus ? "state-focus" : ""
                }`}
            >
                {this.props.prepend ? (
                    <span className="input-group-prepend">
                        {this.props.isPrependText ? (
                            <span className="input-group-text">
                                {this.props.prepend}
                            </span>
                        ) : (
                            this.props.prepend
                        )}
                    </span>
                ) : null}
                <Input
                    type={this.props.type}
                    className="form-control"
                    size={this.props.size}
                    placeholder={this.props.placeholder}
                    {...filterInputProps(this.props)}
                    {...filterEventProps(this.props)}
                    onFocus={this.onChangeFocus.bind(this, true)}
                    onBlur={this.onChangeFocus.bind(this, false)}
                />
                {this.props.append ? (
                    <span className="input-group-prepend">
                        {this.props.isAppendText ? (
                            <span className="input-group-text">
                                {this.props.append}
                            </span>
                        ) : this.props.appendFunc ? (
                            this.props.value !== "" ? (
                                addPropsToChildren(this.props.append, {
                                    onClick: () => this.props.appendFunc(this)
                                })
                            ) : null
                        ) : (
                            this.props.append
                        )}
                    </span>
                ) : null}
            </div>
        );
    }
}
