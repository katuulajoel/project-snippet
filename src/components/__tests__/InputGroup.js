import React from "react";
import renderer from "react-test-renderer";
import InputGroup from "../InputGroup";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";

const middlewares = [thunk];

const mockAppState = {
  Auth: {},
  Invoice: { search: {} },
};

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state || mockAppState);
};
const appendFunc = jest.fn();
const append = <button className="tst">Add</button>;

describe("InputGroup component test", () => {
  it("Snapshot test for InputGroup component", () => {
    const tree = renderer
      .create(
        <Provider store={mockAppStore()}>
          <InputGroup appendFunc={appendFunc} value="test" append={append} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
