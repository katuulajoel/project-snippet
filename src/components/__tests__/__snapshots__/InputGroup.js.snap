// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup component test Snapshot test for InputGroup component 1`] = `
<div
  className="input-group "
>
  <div
    className="sc-bdnxRM iEBWwR"
    data-testid="input"
  >
    <input
      className="form-control"
      data-testid=""
      id=""
      label="Label"
      placeholder="input"
      type="input"
      value="test"
    />
  </div>
  <span
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
    >
      <button
        className="tst"
      >
        Add
      </button>
    </span>
  </span>
</div>
`;
